lib/SIL/SILLocation.cpp

```diff
@@ -211,13 +211,10 @@ MandatoryInlinedLocation::getMandatoryInlinedLocation(SILLocation L) {
   if (Decl *D = L.getAsASTNode<Decl>())
     return MandatoryInlinedLocation(D, L.getSpecialFlags());
 
-  if (L.isSILFile())
-    return MandatoryInlinedLocation(L.Loc.SILFileLoc, L.getSpecialFlags());
-
   if (L.isInTopLevel())
     return MandatoryInlinedLocation::getModuleLocation(L.getSpecialFlags());
 
-  llvm_unreachable("Cannot construct Inlined loc from the given location.");
+  return MandatoryInlinedLocation(L.Loc.SILFileLoc, L.getSpecialFlags());
```

Comment:
That looks a bit strange. Shouldn't `if (L.isSILFile())` get triggered when this is a SILFileLoc?
Loc is a union so I'm afraid that this will just crash later...

Can you reproduce the crash and extract `L.KindData`?